architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,247 @@
+
+   1. **Purpose:** To hash the resource ID to generate a unique key for the resource name in the data store.
+   2. **Library:** Uses the Go standard `crypto/sha1` package: [crypto/sha1](https://pkg.go.dev/crypto/sha1)
+   3. **Type:** SHA-1  
+      _Note:_ SHA-1 is used here for generating unique identifiers, not for security.
+
+3. **Hashing Resource IDs for Tracking Resource Names**: [Link to code](https://github.com/radius-project/radius/blob/main/pkg/ucp/trackedresource/name.go#L52)
+
+   1. **Purpose:** To hash a given resource ID to compute the tracked resource name.
+   2. **Library:** Uses the Go standard `crypto/sha1` package: [crypto/sha1](https://pkg.go.dev/crypto/sha1)
+   3. **Type:** SHA-1  
+      _Note:_ As previously noted, SHA-1 is used for optimization purposes only.
+
+4. **Hashing Input Data to Generate New ETags**: [Link to code](https://github.com/radius-project/radius/blob/main/pkg/ucp/util/etag/etag.go#L30)
+
+   1. **Purpose:** To hash input data to generate new ETags for resource versioning.
+   2. **Library:** Uses the Go standard `crypto/sha1` package: [crypto/sha1](https://pkg.go.dev/crypto/sha1)
+   3. **Type:** SHA-1  
+      _Note:_ SHA-1 usage here is for generating ETags and not for cryptographic security.
+
+---
+
+##### General Note on SHA-1 Usage
+
+In all instances where SHA-1 is utilized within the codebase, it serves for generating hashes to ensure uniqueness or detect changes efficiently. It is **not** used for cryptographic security purposes due to known vulnerabilities. For security-critical hashing, a more secure algorithm (e.g., SHA-256) should be employed.
+
+---
+
+#### Storage of Secrets
+
+Below you will find where and how Radius stores secrets. We create Kubernetes Secret objects and rely on Kubernetes security measures to protect these secrets.
+
+1. **Creating or Updating a Cloud Provider Credential Resource**: [Azure](https://github.com/radius-project/radius/blob/95409fe179d7adca884a3fc1d82f326bc81c8da0/pkg/ucp/frontend/controller/credentials/azure/createorupdateazurecredential.go#L89), [AWS](https://github.com/radius-project/radius/blob/95409fe179d7adca884a3fc1d82f326bc81c8da0/pkg/ucp/frontend/controller/credentials/aws/createorupdateawscredential.go#L81). When a **Cloud Provider Credential** resource is deployed, Radius creates a new Kubernetes Secret or updates an existing one. The Kubernetes Secret is deleted when the **Cloud Provider Credential** is deleted.
+
+#### Data Serialization / Formats
+
+We use custom parsers to parse Radius-related resource IDs and do not use any other custom parsers and instead rely on Kubernetes built-in parsers. Therefore, we trust Kubernetes security measures to handle data serialization and formats securely. The custom parser that parses Radius resource IDs has its own security mechanisms that don't accept anything other than a Radius resource ID.
```

Comment:
Was this copied from the controller threat model?

I don't think we're using Kubernetes to parse JSON.

Reply:
Yes, this was copied from the Controller Threat Model. I will update this part accordingly.

Reply:
1st: Resource ID parser
2nd: Go standard library
All data interchanges are done using JSON and we use GO Standard library..